Src/Particle/AMReX_ParIter.H

```diff
@@ -65,18 +65,28 @@ public:
 #ifdef AMREX_USE_OMP
     void operator++ ()
     {
-        if (dynamic) {
+        bool has_particles = false;
+        while (isValid() && !has_particles) {
+            if (dynamic) {
 #pragma omp atomic capture
-            m_pariter_index = nextDynamicIndex++;
-        } else {
-            ++m_pariter_index;
+                m_pariter_index = nextDynamicIndex++;
+            } else {
+                ++m_pariter_index;
```

Comment:
Shouldn't this line be an atomic operation?

Reply:
(I mean the `++m_pariter_index`)

Reply:
It's a thread local variable. No atomic op needed.

Reply:
Ah right! Sorry, for some reason I thought that that member variable was shared, but actually the whole object is thread-local, right? Atomic op is needed only for `nextDynamicIndex` because that is static class variable.